### PR TITLE
support for auto axes display in char.js. Closes #38972

### LINKED ISSUE #38972: Display property of CommonAxe should be Boolean|String
In the following declaration

```
interface CommonAxe {
        bounds?: string;
        type?: ScaleType | string;
        display?: boolean;
        id?: string;
        stacked?: boolean;
        position?: string;
        ticks?: TickOptions;
        gridLines?: GridLineOptions;
        barThickness?: number | "flex";
        maxBarThickness?: number;
        minBarLength?: number;
        scaleLabel?: ScaleTitleOptions;
        time?: TimeScale;
        offset?: boolean;
        beforeUpdate?(scale?: any): void;
        beforeSetDimension?(scale?: any): void;
        beforeDataLimits?(scale?: any): void;
        beforeBuildTicks?(scale?: any): void;
        beforeTickToLabelConversion?(scale?: any): void;
        beforeCalculateTickRotation?(scale?: any): void;
        beforeFit?(scale?: any): void;
        afterUpdate?(scale?: any): void;
        afterSetDimension?(scale?: any): void;
        afterDataLimits?(scale?: any): void;
        afterBuildTicks?(scale?: any): void;
        afterTickToLabelConversion?(scale?: any): void;
        afterCalculateTickRotation?(scale?: any): void;
        afterFit?(scale?: any): void;
    }
```
Display should be allowed to have 'Auto', true and false values. However, since it is declared as boolean, it is giving compile error when used in Angular projects.

Code snippet
```
import {ChartDataSets, ChartData, ChartOptions} from 'chart.js'; 
import {Color, BaseChartDirective, Label} from 'ng2-charts';'
```
```
private lineChartOptions : (ChartOptions) = {
    responsive: true,
    scales:{
      xAxes: [{
        id: 'x-axis-0',
        position: 'bottom'
      }],
      yAxes: [
        {
          id: 'y-axis-0',
          display: 'auto' as any,
          position: 'left'
        },
        {
          id: 'y-axis-1',
          display: 'auto' as any,
          position: 'right',
          gridLines: {
            color: 'rgba(255,0,0,0.3)',
          },
          ticks: {
            fontColor: 'red',
          }
        },        
      ]
    }
  }
```
However this property type is boolean|string in chartjs
https://www.chartjs.org/docs/latest/axes/

### PR DESCRIPTION
**Pull request by vidi42**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://www.chartjs.org/docs/latest/axes/
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/chart.js/chart.js-tests.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { BorderWidth, Chart, ChartData, Point, ChartColor } from 'chart.js';
+import { BorderWidth, Chart, ChartColor, ChartData, ChartOptions, Point } from "chart.js";
```
Comment (ElianCordoba) [inline types/chart.js/chart.js-tests.ts]: Keep the imports with single quotes

**Review by ElianCordoba — APPROVED** [review]
Besides the small change I mentioned LGTM

File: types/chart.js/chart.js-tests.ts [hunk 2]
```diff
@@ -241,3 +241,31 @@ const chartWithScriptedOptions = new Chart(new CanvasRenderingContext2D(), {
...
+    responsive: true,
+    scales: {
+        xAxes: [{
+            id: 'x-axis-0',
+            position: 'bottom'
+        }],
+        yAxes: [
+            {
+                id: 'y-axis-0',
+                display: 'auto' as any,
```
Comment (danmana) [inline types/chart.js/chart.js-tests.ts]: with the new ts definition the cast `as any` should not be needed

```suggestion
                display: 'auto',
```

File: types/chart.js/chart.js-tests.ts [hunk 3]
```diff
@@ -241,3 +241,31 @@ const chartWithScriptedOptions = new Chart(new CanvasRenderingContext2D(), {
...
+                position: 'left'
+            },
+            {
+                id: 'y-axis-1',
+                display: 'auto' as any,
```
Comment (danmana) [inline types/chart.js/chart.js-tests.ts]:
```suggestion
                display: 'auto',
```

**Review by danmana — CHANGES_REQUESTED** [review]

**Pull request closed without merging**